### ISSUE TITLE
Share extension sometimes fails without feedback

### ISSUE THREAD

**Issue by kiwijam** [xmollv/Llitgi#69]
In particular I have noticed that sharing from the Shazam and Inoreader apps can fail without any indication as to why.

I believe this might be because Llitgi tries to add something that isn't an URL to Pocket.

**Comment by xmollv** [comment]
Thanks for reporting it. It's a 100% of the time on Shazam? I'll give it a try when I get back home from WWDC 👍🏼

**Comment by kiwijam** [comment]
Tried with 15 songs just now - seems to be 100%, yes.

I also noticed that the 2nd time I try to share from Shazam to Llitgi, Shazam's UI will completely freeze - not sure if that's a Shazam or Llitgi problem.

**Comment by xmollv** [comment]
@kiwijam Can you show me where you're triggering the share? Because if I look for a song, tap share and then drop it onto Llitgi, what I'm receiving is this:

```
(lldb) po sharedText
"I used Shazam to discover Boom by X Ambassadors."
```

I'm seeing that if I drop it into Mail it seems to have the link after the text, but I'm not being able to grab it.

**Comment by xmollv** [comment]
It got automatically closed 🤦‍♂

**Comment by kiwijam** [comment]
I'm not sure how applicable this is to Llitgi, but what I had to do for Shazam and Inoreader to work with a `Scriptable' script was to iterate over all inputs shared and test for the presence of a URL:

Pseudo-code:
```
for (array of [args.urls, args.fileURLs, args.plainTexts]) {
  for (item of array) {
    if (/^https?:\/\//.test(item)) {
      // found url
    }
  }
}
```

**Comment by xmollv** [comment]
@kiwijam With the code that I added to fix the YouTube issue it should be enough, because the `NSDataDetector` should be able to get the `URL` with no issues. The problem that I'm having is that the `ItemProvider` -no matter how I try to grab it- never has the `URL`, just the text.

https://twitter.com/xmollv/status/1138099590970904579

When I have some free time I'll try to dig deeper.

**Comment by xmollv** [comment]
Okay found the issue with Shazam. I'm just handling the first attachment on the extension, instead of all of them... I need to think of a decent way to refactor this in order to loop over every extension and just grab the ones that are `URL`s. It also needs to support multiple requests to the Pocket API which it doesn't right now.

https://github.com/xmollv/Llitgi/blob/6ab62692f6972a1554110e5b5b8f6056e24c785e/ShareExtension/ShareViewController.swift#L64

```
(lldb) po (self.extensionContext?.inputItems.first as? NSExtensionItem)?.attachments
▿ Optional<Array<NSItemProvider>>
  ▿ some : 2 elements
    - 0 : <NSItemProvider: 0x2835f6370> {types = (
    "public.plain-text"
)}
    - 1 : <NSItemProvider: 0x2835f64c0> {types = (
    "public.url"
)}
```

**Comment by xmollv** [comment]
Fixed in #71. @kiwijam I'll push the update in a moment, once it's live on the store let me know if it still doesn't work for you, it seems fine to me with the testing that I've done here.
